src/routes/products/[id([0-9]+)]/edit.svelte

```diff
@@ -66,13 +66,23 @@
   let errorMessage = null;
   let warningDialogOpen = false;
 
-  async function addColor(color) {
-    let resp = await api.post('/colors', { data: { value: color } });
-    if (resp.status == 200) {
+  let colorDebounce = false;
+  async function addColor({ detail: color }) {
+    if (colorDebounce || colors.include(color)) {
+      return;
+    }
+
+    try {
+      await api.json(api.post(
+        '/colors',
+        { data: { value: color }, csrfToken: account.csrf_token },
+      ));
       colors.push({ value: color });
       colors = colors;
-    } else {
-      console.error(`Color creation failed (${resp.status})`);
+      colorDebounce = true;
+      setTimeout(() => colorDebounce = false, 200);
```

Comment:
I couldn't reproduce Max's browser's behavior (of firing continuously), but it seems to me that this code implements throttle and not debounce

Reply:
Dammit, not again. Yeah, I couldn't reproduce the thing either, it's probably just the Mac's system color picker.